core/src/scene/scene.cpp

```diff
@@ -70,6 +70,21 @@ Scene::Scene(const Scene& _other)
 
 Scene::~Scene() {}
 
+bool Scene::removeGlobalUpdate(const std::string& key) {
+    bool found = false;
+    for (auto& globalUpdate : m_referencedGlobals) {
+        if (globalUpdate.first == key) {
+            globalUpdate.swap(m_referencedGlobals[m_referencedGlobals.size() - 1]);
```

Comment:
I like this trick :) Slightly more concise would be `globalUpdate.swap(m_referencedGlobals.back());`

Also maybe `globalRef` would make a better name for this variable.